app/src/main/AndroidManifest.xml

```diff
@@ -49,6 +49,12 @@
             android:enabled="true" />
 
         <receiver android:name=".Receivers.NotificationAlarmReceiver" />
+        <receiver android:name=".Receivers.NetworkDialogHandler">
```

Comment:
I'd done exactly that but I guess dynamic registration should be avoided or something, no? Most people do it in the manifest or so.